Client/src/components/game.tsx

```diff
@@ -0,0 +1,127 @@
+import '../css/game.css';
+import Maze from './maze';
+import GameStats from './gameStats';
+import ToolBar from './toolbar';
+import { useEffect, useState } from 'react';
+import { minionType } from '../utils/types';
+import { Graph, value } from '../utils/graph';
+import { bFS, dijkstra, getDirection, vBFS } from '../utils/path-finding-algo';
+
+function Game() {
+
+  const [boxSize, setBoxSize] = useState(30);
+  const [mazeCompleted, setMazeCompleted] = useState(false);
+  const [minions, setMinions] = useState<{[key: number]: minionType}>({
+    0: {id: 0, xPos: 1, yPos: 0, rotation: 0}, 
+    1: {id: 1, xPos: 1, yPos: 0, rotation: 0}, 
+    2: {id: 2, xPos: 1, yPos: 0, rotation: 0}, 
+    3: {id: 3, xPos: 1, yPos: 0, rotation: 0}, 
+    4: {id: 4, xPos: 2, yPos: 4, rotation: 0}
+  });
+  const [currentMinion, setCurrentMinion] = useState<null | number>(null);
+  const [currentTile, setCurrentTile] = useState<null | {xPos:number, yPos:number}>(null);
+  const [waitingForTile, setWaitingForTile] = useState(false);
+  const [currentGraph, setCurrentGraph] = useState<Graph>();
+  const [height, setHeight] = useState(48);
+  const [width, setWidth] = useState(72);
+  const [movingMinions, setMovingMinions] = useState<number[]>([]);
+  const speed = 200;
+  const minBoxSize = 5;
+  const maxBoxSize = 100;
+
+  function addNewMinion() {
+    const newId = Object.keys(minions).length;
+    setMinions(prevMinions => {
+      return {
+        ...prevMinions,
+        [newId]: {
+          id: newId,
+          xPos: 0,
+          yPos: 0,
+          rotation: 0
+        }
+      }
+    })
+  }
+
+  useEffect(() => {
+    console.log({minions});
+  }, [minions])
+
+  useEffect(() => {
+    if (currentMinion !== null) {
+      console.log(currentMinion);
+      setCurrentTile(null);
+      setWaitingForTile(true);
+    } else {
+      setWaitingForTile(false);
+    }
+  }, [currentMinion])
+
+  useEffect(() => {
```

Comment:
this should be add in a separate file, maybe in a functionalities folder and made a function.

Reply:
👍